### ISSUE TITLE
Reinstate smoketest.rsocket.SampleRSocketApplicationTests.rSocketEndpoint()

### ISSUE THREAD

**Issue by wilkinsona** [spring-projects/spring-boot#22725]
It fails at the moment:

```
java.lang.AssertionError: expectation "assertNext" failed (expected: onNext(); actual: onError(ConnectionErrorException (0x101): Unexpected error during frame handling))
	at reactor.test.MessageFormatter.assertionError(MessageFormatter.java:115)
	at reactor.test.MessageFormatter.failPrefix(MessageFormatter.java:104)
	at reactor.test.MessageFormatter.fail(MessageFormatter.java:73)
	at reactor.test.MessageFormatter.failOptional(MessageFormatter.java:88)
	at reactor.test.DefaultStepVerifierBuilder.lambda$consumeNextWith$1(DefaultStepVerifierBuilder.java:268)
	at reactor.test.DefaultStepVerifierBuilder$SignalEvent.test(DefaultStepVerifierBuilder.java:2213)
	at reactor.test.DefaultStepVerifierBuilder$DefaultVerifySubscriber.onSignal(DefaultStepVerifierBuilder.java:1485)
	at reactor.test.DefaultStepVerifierBuilder$DefaultVerifySubscriber.onExpectation(DefaultStepVerifierBuilder.java:1433)
	at reactor.test.DefaultStepVerifierBuilder$DefaultVerifySubscriber.onError(DefaultStepVerifierBuilder.java:1092)
	at reactor.core.publisher.FluxMap$MapSubscriber.onError(FluxMap.java:132)
	at reactor.core.publisher.FluxMap$MapSubscriber.onError(FluxMap.java:132)
	at io.rsocket.core.ResolvingOperator$DeferredResolution.onError(ResolvingOperator.java:476)
	at io.rsocket.core.RequestResponseRequesterMono.handleError(RequestResponseRequesterMono.java:245)
	at io.rsocket.core.RSocketRequester.lambda$terminate$4(RSocketRequester.java:343)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at io.rsocket.core.RSocketRequester.terminate(RSocketRequester.java:340)
	at io.rsocket.core.RSocketRequester.tryTerminate(RSocketRequester.java:317)
	at io.rsocket.core.RSocketRequester.tryTerminateOnZeroError(RSocketRequester.java:310)
	at io.rsocket.core.RSocketRequester.handleStreamZero(RSocketRequester.java:221)
	at io.rsocket.core.RSocketRequester.handleIncomingFrames(RSocketRequester.java:203)
	at reactor.core.publisher.LambdaSubscriber.onNext(LambdaSubscriber.java:160)
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyInner.onNext(MonoFlatMapMany.java:250)
	at reactor.core.publisher.FluxGroupBy$UnicastGroupedFlux.drainRegular(FluxGroupBy.java:561)
	at reactor.core.publisher.FluxGroupBy$UnicastGroupedFlux.drain(FluxGroupBy.java:637)
	at reactor.core.publisher.FluxGroupBy$UnicastGroupedFlux.subscribe(FluxGroupBy.java:703)
	at reactor.core.publisher.Flux.subscribe(Flux.java:7961)
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain.onNext(MonoFlatMapMany.java:195)
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1782)
	at reactor.core.publisher.MonoProcessor.onNext(MonoProcessor.java:317)
	at io.rsocket.internal.ClientServerInputMultiplexer.lambda$new$1(ClientServerInputMultiplexer.java:126)
	at reactor.core.publisher.LambdaSubscriber.onNext(LambdaSubscriber.java:160)
	at reactor.core.publisher.FluxGroupBy$GroupByMain.drainLoop(FluxGroupBy.java:387)
	at reactor.core.publisher.FluxGroupBy$GroupByMain.drain(FluxGroupBy.java:323)
	at reactor.core.publisher.FluxGroupBy$GroupByMain.onNext(FluxGroupBy.java:207)
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:120)
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:120)
	at reactor.netty.channel.FluxReceive.drainReceiver(FluxReceive.java:260)
	at reactor.netty.channel.FluxReceive.onInboundNext(FluxReceive.java:366)
	at reactor.netty.channel.ChannelOperations.onInboundNext(ChannelOperations.java:373)
	at reactor.netty.channel.ChannelOperationsHandler.channelRead(ChannelOperationsHandler.java:94)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:714)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
	Suppressed: ConnectionErrorException (0x101): Unexpected error during frame handling
		at io.rsocket.exceptions.Exceptions.from(Exceptions.java:66)
		at io.rsocket.core.RSocketRequester.lambda$tryTerminateOnZeroError$3(RSocketRequester.java:310)
		at io.rsocket.core.RSocketRequester.tryTerminate(RSocketRequester.java:315)
		... 44 more
```

@rwinch is investigating the security side of things.

**Comment by bclozel** [comment]
This is blocked by spring-projects/spring-security#8948, I'm rescheduling this issue to the next milestone as a result.

**Comment by bclozel** [comment]
Waiting on #23225

**Comment by snicoll** [comment]
Closing in favour of PR #23246
